### ISSUE TITLE
Build fails: can't find C standard library headers

### ISSUE THREAD

**Issue by ericfischer** [mapbox/node-cpp-skel#135]
The build fails for me:

```
➤ git clone git@github.com:mapbox/node-cpp-skel.git
…
➤ cd node-cpp-skel
➤ make
…
  /Users/enf/node-cpp-skel/mason_packages/.link/bin/clang++ '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' '-DNDEBUG' -I/Users/enf/.node-gyp/8.4.0/src -I/Users/enf/.node-gyp/8.4.0/deps/uv/include -I/Users/enf/.node-gyp/8.4.0/deps/v8/include  -O3 -mmacosx-version-min=10.8 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fvisibility-inlines-hidden -fno-threadsafe-statics -isystem /Users/enf/node-cpp-skel/node_modules/nan -isystem /Users/enf/node-cpp-skel/mason_packages/.link/include/ -Wall -Wextra -Weffc++ -Wconversion -pedantic-errors -Wconversion -Wshadow -Wfloat-equal -Wuninitialized -Wunreachable-code -Wold-style-cast -Wno-error=unused-variable -Werror -flto -flto -MMD -MF ./Release/.deps/Release/obj.target/module/src/module.o.d.raw  -c -o Release/obj.target/module/src/module.o ../src/module.cpp
In file included from ../src/module.cpp:1:
In file included from ../src/object_async/hello_async.hpp:2:
In file included from /Users/enf/node-cpp-skel/node_modules/nan/nan.h:46:
In file included from /Users/enf/.node-gyp/8.4.0/deps/uv/include/uv.h:48:
/Users/enf/.node-gyp/8.4.0/deps/uv/include/uv-errno.h:25:10: fatal error: 'errno.h' file not found
#include <errno.h>
         ^~~~~~~~~
1 error generated.
make[1]: *** [Release/obj.target/module/src/module.o] Error 1
```

The compiler from Mason can't find my `<errno.h>` header.

It works if I use my local installation of clang-5.0.1:

```
➤ export CXX=/Users/enf/clang+llvm-5.0.1-final-x86_64-apple-darwin/bin/clang++
➤ export LINK="./gyp-mac-tool flock ./linker.lock /Users/enf/clang+llvm-5.0.1-final-x86_64-apple-darwin/bin/clang++"
➤ make
…
info Completed
  ./gyp-mac-tool flock ./linker.lock /Users/enf/clang+llvm-5.0.1-final-x86_64-apple-darwin/bin/clang++ -bundle -Wl,-bind_at_load -Wl,-no_pie -Wl,-search_paths_first -flto -flto -Wl,-dead_strip -mmacosx-version-min=10.8 -arch x86_64 -L./Release  -o /Users/enf/node-cpp-skel/lib/binding/module.node Release/obj.target/module/src/module.o Release/obj.target/module/src/standalone/hello.o Release/obj.target/module/src/standalone_async/hello_async.o Release/obj.target/module/src/object_sync/hello.o Release/obj.target/module/src/object_async/hello_async.o -undefined dynamic_lookup
  SOLINK_MODULE(target) /Users/enf/node-cpp-skel/lib/binding/module.node: Finished
  rm -rf "Release/module.node" && cp -af "/Users/enf/node-cpp-skel/lib/binding/module.node" "Release/module.node"
run 'make clean' for full rebuild
```

This is on MacOS X 10.11.6,

```
➤ uname -a
Darwin C02LP330F6T6 15.6.0 Darwin Kernel Version 15.6.0: Mon Nov 13 21:58:35 PST 2017; root:xnu-3248.72.11~1/RELEASE_X86_64 x86_64 i386 MacBookAir6,2 Darwin
➤ c++ --version
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Maybe it also needs to install libc through Mason? I don't know how to configure that though.

**Comment by springmeyer** [comment]
This is likely happening because of this bug: https://github.com/mapbox/mason-js/issues/58#issuecomment-398586803. Basically the mason clang++ depends on the XCode C++ headers and they are getting deleted, inadvertently, by mason-js. Aiming to solve in #141.

**Comment by springmeyer** [comment]
worked around this problem for now in #146

**Comment by springmeyer** [comment]
Solved by #152
